WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m

```diff
@@ -473,6 +486,37 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForGeneralSettingsIn
             [self.languageTextCell setTextValue:name];
             return self.languageTextCell;
         }
+        case SiteSettingsGeneralTimeZone:
+        {
+            NSString *timeZoneString;
+            if (self.blog.settings.timeZoneString != nil) {
+                timeZoneString = self.blog.settings.timeZoneString;
+            } else {
+                // timeZoneString is nil, so that means we can't fetch settings for this blog
+                // so instead we'll use the blog options to get the timezone
+                timeZoneString = [self.blog getOptionValue:@"timezone"];
+            }
+            if ([timeZoneString isEqualToString:@""]) {
+                NSNumber *gmtOffset;
+                if (self.blog.settings.gmtOffset != nil) {
+                    gmtOffset = self.blog.settings.gmtOffset;
+                } else {
+                    gmtOffset = [self.blog getOptionValue:@"gmt_offset"];
+                }
+                long hoursUTC = gmtOffset.integerValue;
+                long minutesUTC = fabs(([gmtOffset doubleValue] - hoursUTC) * 60);
+                NSString *utcString;
+                if (minutesUTC == 0) {
+                    utcString = [NSString stringWithFormat:@"UTC %+ld", hoursUTC];
```

Comment:
I think ideally we should avoid doing this time formatting ourselves - we can use `DateComponents` and `DateComponentsFormatter` to produce a string of the same format, which will take into account any internationalization differences.

A rough Swift demo:

```
var components = DateComponents()
components.hour = ...
components.minute = ...

let formatter = DateComponentsFormatter()
formatter.allowedUnits = [.hour, .minute]
formatter.zeroFormattingBehavior = .pad
formatter.string(from: components)
```